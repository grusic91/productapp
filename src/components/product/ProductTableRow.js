import React from 'react';

export class ProductTableRow extends React.Component {
    render() {
        let p = this.props.product;
        if(p) {
            return <tr>
            <td>{ p.id }</td>
            <td>{ p.name }</td>
            <td>{ p.category}</td>
            <td className="text-right">${ Number(p.price).toFixed(2) }</td>
            <td>
                <button 
                    className="btn btn-sm btn-warning m-1"
                    onClick={ () => this.props.editCallback(p) }
                > 
                    Edit 
                </button>
                <button
                    className="btn btn-sm btn-danger m-1"
                    onClick={ () => this.props.deleteCallback(p) }
                >
                    Delete
                </button>
            </td>
        </tr>
        }
        
        return <h1>Loading...</h1>
    }
}
