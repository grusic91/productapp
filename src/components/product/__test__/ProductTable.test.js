import React from 'react';
import { shallow } from 'enzyme';
import { ProductTable } from '../ProductTable';
import { ProductTableRow } from '../ProductTableRow';

describe('<ProductTable />', () => {
    let wrapper;
    beforeAll(() => wrapper = shallow(<ProductTable 
        products={[]}
        editCallback={jest.fn()}
        deleteCallback={jest.fn()}
    />));

    it("renders Components properly", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it("renders one table element", () => {        
        expect(wrapper.find('table').length).toEqual(1);
    });

    it("find th element where text is Products", () => {
        expect(wrapper.find('th').first().text()).toEqual("Products");
    });

    it("renders all matching headers of columns", () => {
        expect(wrapper.containsMatchingElement(
            <tr>
                <th>ID</th><th>Name</th><th>Category</th><th>Price</th>
            </tr>
        )).toEqual(true);
    });

    it("renders tbody element", () => {
        expect(wrapper.find('tbody')).toHaveLength(1);
    });

    it("in case there is no products, ProductTableRow is not rendered", () => {
        expect(wrapper.containsMatchingElement(<ProductTableRow />)).toEqual(false);
    });

    it("in case we have two products ProductTable rneders two ProductTableRow components", () => {
        // populate products props on ProductTable
        wrapper.setProps({
            products: [
                {
                    id: 1,
                    name: "Kayak1",
                    category: "Watersports",
                    price: "275"
                },
                {
                    id: 2,
                    name: "Lifejacket",
                    category: "Watersports",
                    price: "48.95"
                }
            ]
        });
        expect(wrapper.find(ProductTableRow)).toHaveLength(2);
    });
});
