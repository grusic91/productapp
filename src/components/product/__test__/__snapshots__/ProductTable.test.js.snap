// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTable /> renders Components properly 1`] = `ShallowWrapper {}`;
